### PR TITLE
[REVIEW] Update container and notebook, add mlflow

### PR DESCRIPTION
**Pull request by zronaghi**
- Updated RAPIDS container to 0.16 stable
- Added CV-folds and multi-GPU to argparse
- Added MLflow logging

### CODE REVIEW

**Review by miroenev — COMMENTED** [review]
Looks great! Just a few small comments. 

p.s. @wphicks convinced me to swap all print statements with a logger so we can control output levels

File: azure/code/train_rapids.py [hunk 1]
```diff
@@ -87,44 +93,45 @@ def main():
...
 
         # split data
         X_train, X_test, y_train, y_test, _ = azure_ml.split_data(X, y, random_state=i_train_fold)
-        # train model 
-        trained_model, training_time = azure_ml.train_model (X_train, y_train, model_params)
+        # train model
+        trained_model, training_time = azure_ml.train_model(X_train, y_train, model_params)
 
-        train_time_per_fold += [ round( training_time, 4) ]
+        train_time_per_fold += [round(training_time, 4)]
```
[the remark this answers is not in the document]
Comment (zronaghi) [reply azure/code/train_rapids.py]: Sounds good, replaced with append

File: azure/code/train_rapids.py [hunk 2]
```diff
@@ -87,44 +93,45 @@ def main():
...
-    run.log('Accuracy', np.float(global_best_test_accuracy))
-    print( '\n Accuracy             :', global_best_test_accuracy)
-    print( '\n accuracy per fold    :', accuracy_per_fold)
-    print( '\n train-time per fold  :', train_time_per_fold)
-    print( '\n train-time all folds  :', sum(train_time_per_fold))
-    print( '\n infer-time per fold  :', infer_time_per_fold)
-    print( '\n infer-time all folds  :', sum(infer_time_per_fold))
+    mlflow.log_metric('Total training inference time', np.float(training_time + infer_time))
+    mlflow.log_metric('Accuracy', np.float(global_best_test_accuracy))
+    print('\n Accuracy             :', global_best_test_accuracy)
```
[the remark this answers is not in the document]
Comment (zronaghi) [reply azure/code/train_rapids.py:124]: I think print will handle the conversion for different OS

[the remark this answers is not in the document]
Comment (wphicks) [reply azure/code/train_rapids.py:124]: Yep, you're right @zronaghi! Or to be more specific, the current `sys.stdout` stream controls how newline characters are dealt with, and by default, it's set up to provide the correct conversion for the current OS.

**Comment by miroenev** [comment]
Looks great, please merge!

**Pull request merged**